### PR TITLE
Make a truly async model pipeline work for WebApi publisher.

### PR DESCRIPTION
**Pull request by rayao**
A truly async model pipeline leads to unsuccessful routing registration. We didn't see this in our existing samples because their model pipelines are synchronous in nature.
This change tries to defer the actual RESTier routing setup until the model pipeline returns. Before that, requests run on a placeholder routing.

### CODE REVIEW

**Comment by rayao** [comment]
#304

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs [hunk 1]
```diff
@@ -44,20 +47,24 @@ public static class HttpConfigurationExtensions
...
+                    ODataRouteConstants.Batch : (routePrefix + '/' + ODataRouteConstants.Batch);
+                routes.MapHttpBatchRoute(routeName + "Batch", routeTemplate, batchHandler);
+            }
 
-                if (batchHandler != null && batchHandler.ApiFactory == null)
-                {
-                    batchHandler.ApiFactory = apiFactory;
-                }
+            var pathHandler = new DefaultODataPathHandler();
+            ////defaultODataPathHandler.ResolverSetttings = config.GetResolverSettings();
```
Comment (rayao) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: ResolverSettings is an internal member, must find a way to set it if no other convenient API.

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs [hunk 2]
```diff
@@ -115,5 +122,79 @@ private static void EnsurePayloadValueConverter(this IEdmModel model)
...
+            public Func<IApi> ApiFactory { get; private set; }
+
+            public IEdmModel FinalModel { get; private set; }
+
+            public RestierRouteConstraint(IODataPathHandler pathHandler, string routeName, Func<IApi> apiFactory)
+            {
+                var conventions = ODataRoutingConventions.CreateDefault();
+                Placeholder = new ODataPathRouteConstraint(pathHandler, new EdmModel(), routeName, conventions);
+                ApiFactory = apiFactory;
+                Task.Delay(5000).ContinueWith(_ => TryInitModel());
```
Comment (rayao) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: Sorry the Delay is for testing, to make the pipeline truly async. It's not required.

**Comment by rayao** [comment]
I'm not quite sure if we should go this way. Another option is to for the model pipeline synchronous.

**Comment by lewischeng-ms** [comment]
@rayao, based on our offline discussion, please change the PR accordingly and merge. Thanks！

**Pull request closed without merging**